plugins/controlboard/src/ControlBoard.cpp

```diff
@@ -132,6 +132,25 @@ void ControlBoard::Configure(const Entity& _entity,
         return;
     }
 
+    auto& coupling_group_bottle = m_pluginParameters.findGroup("COUPLING");
+    if (!coupling_group_bottle.isNull()) {
+        m_pluginParameters.unput("device");
```

Comment:
Can we avoid to change the `m_pluginParameters` attribute? If we need to get all the parameters of `m_pluginParameters`, we can copy this to a `m_couplingParameters` attribute, and then set the device just there. However, just to understand, why do we need to pass the whole plugin parameters to the coupling device?

Reply:
Ack, can you add a comment explaining that and avoid to change the content of `m_pluginParameters` and create a new `yarp::os::Property` instance instead? Thanks!